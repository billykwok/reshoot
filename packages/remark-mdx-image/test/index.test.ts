import { compile } from '@mdx-js/mdx';
import { describe, expect, test } from '@jest/globals';
import { format } from 'prettier';

import reshootRemarkMdxImage from '../src';

expect.addSnapshotSerializer({
  serialize: (value: string) => format(value, { parser: 'babel' }),
  test: () => true,
});

describe('MDX with @reshoot/remark-mdx-image', () => {
  test('should correctly transform when default settings', async () => {
    const result = await compile(
      `
        ## Hellow World

        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.

        ![](https://example.com/image.png){{color:'#f4f4f4'}}
        ![Image alt text 1](assets/image.png "Image title 1")

        Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        
        <div id="abc">
          ![](../assets/image.png)
          ![](../assets/image.png "Image title 2"){{color:'#f4f4f4'}}
          ![Image alt text 2](../assets/image.png){{color:'#f4f4f4'}}
        </div>

        Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.

        [![Image alt text 3](../assets/image.png?reshoot "Image title 3"){{color:'#f4f4f4'}}](https://example.com "Link title")

        Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      `,
      {
        jsx: true,
        remarkPlugins: [reshootRemarkMdxImage],
      }
    );
    expect(result.toString()).toMatchSnapshot();
  });

  test('should correctly transform when relativeToDocument is disabled', async () => {
    const result = await compile(
      `
        ## Hellow World

        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.

        ![](https://example.com/image.png){{color:'#f4f4f4'}}
        ![Image alt text 1](assets/image.png "Image title 1")

        Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        
        <div id="abc">
          ![](../assets/image.png)
          ![](../assets/image.png "Image title 2"){{color:'#f4f4f4'}}
          ![Image alt text 2](../assets/image.png){{color:'#f4f4f4'}}
        </div>

        Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.

        [![Image alt text 3](../assets/image.png?reshoot "Image title 3"){{color:'#f4f4f4'}}](https://example.com "Link title")

        Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      `,
      {
        jsx: true,
        remarkPlugins: [[reshootRemarkMdxImage, { relativeToDocument: false }]],
      }
    );
    expect(result.value).toMatchSnapshot();
  });

  test('should correctly transform when removeTrailingLineBreak is disabled', async () => {
    const result = await compile(
      `
        ## Hellow World

        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.

        ![](https://example.com/image.png){{color:'#f4f4f4'}}
        ![Image alt text 1](assets/image.png "Image title 1")

        Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        
        <div id="abc">
          ![](../assets/image.png)
          ![](../assets/image.png "Image title 2"){{color:'#f4f4f4'}}
          ![Image alt text 2](../assets/image.png){{color:'#f4f4f4'}}
        </div>

        Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.

        [![Image alt text 3](../assets/image.png?reshoot "Image title 3"){{color:'#f4f4f4'}}](https://example.com "Link title")

        Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      `,
      {
        jsx: true,
        remarkPlugins: [
          [reshootRemarkMdxImage, { removeTrailingLineBreak: false }],
        ],
      }
    );
    expect(result.value).toMatchSnapshot();
  });
});
