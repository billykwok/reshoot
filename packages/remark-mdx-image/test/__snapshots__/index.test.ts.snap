// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MDX with @reshoot/remark-mdx-image should correctly transform when default settings 1`] = `
/*@jsxRuntime automatic @jsxImportSource react*/
import __0___assets_image_png__ from "./assets/image.png";
import __1____assets_image_png__ from "../assets/image.png";
import __2____assets_image_png_color__23f4f4f4__ from "../assets/image.png?color=%23f4f4f4";
import __3____assets_image_png_reshoot_color__23f4f4f4__ from "../assets/image.png?reshoot&color=%23f4f4f4";
function _createMdxContent(props) {
  const _components = Object.assign(
    {
      h2: "h2",
      p: "p",
      img: "img",
      a: "a",
    },
    props.components
  );
  return (
    <>
      <_components.h2>{"Hellow World"}</_components.h2>
      {"\\n"}
      <_components.p>
        {
          "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
        }
      </_components.p>
      {"\\n"}
      <_components.img
        src={"https://example.com/image.png?color=%23f4f4f4"}
        alt=""
      />
      {"\\n"}
      <_components.img
        alt="Image alt text 1"
        meta={__0___assets_image_png__}
        title="Image title 1"
      />
      {"\\n"}
      <_components.p>
        {
          "Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat."
        }
      </_components.p>
      {"\\n"}
      <div id="abc">
        <_components.img alt="" meta={__1____assets_image_png__} />
        <_components.img
          alt=""
          meta={__2____assets_image_png_color__23f4f4f4__}
          title="Image title 2"
        />
        <_components.img
          alt="Image alt text 2"
          meta={__2____assets_image_png_color__23f4f4f4__}
        />
      </div>
      {"\\n"}
      <_components.p>
        {
          "Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
        }
      </_components.p>
      {"\\n"}
      <_components.p>
        <_components.a href="https://example.com" title="Link title">
          <_components.img
            alt="Image alt text 3"
            meta={__3____assets_image_png_reshoot_color__23f4f4f4__}
            title="Image title 3"
          />
        </_components.a>
      </_components.p>
      {"\\n"}
      <_components.p>
        {
          "Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
        }
      </_components.p>
    </>
  );
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = props.components || {};
  return MDXLayout ? (
    <MDXLayout {...props}>
      <_createMdxContent {...props} />
    </MDXLayout>
  ) : (
    _createMdxContent(props)
  );
}
export default MDXContent;

`;

exports[`MDX with @reshoot/remark-mdx-image should correctly transform when relativeToDocument is disabled 1`] = `
/*@jsxRuntime automatic @jsxImportSource react*/
import __0_assets_image_png__ from "assets/image.png";
import __1____assets_image_png__ from "../assets/image.png";
import __2____assets_image_png_color__23f4f4f4__ from "../assets/image.png?color=%23f4f4f4";
import __3____assets_image_png_reshoot_color__23f4f4f4__ from "../assets/image.png?reshoot&color=%23f4f4f4";
function _createMdxContent(props) {
  const _components = Object.assign(
    {
      h2: "h2",
      p: "p",
      img: "img",
      a: "a",
    },
    props.components
  );
  return (
    <>
      <_components.h2>{"Hellow World"}</_components.h2>
      {"\\n"}
      <_components.p>
        {
          "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
        }
      </_components.p>
      {"\\n"}
      <_components.img
        src={"https://example.com/image.png?color=%23f4f4f4"}
        alt=""
      />
      {"\\n"}
      <_components.img
        alt="Image alt text 1"
        meta={__0_assets_image_png__}
        title="Image title 1"
      />
      {"\\n"}
      <_components.p>
        {
          "Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat."
        }
      </_components.p>
      {"\\n"}
      <div id="abc">
        <_components.img alt="" meta={__1____assets_image_png__} />
        <_components.img
          alt=""
          meta={__2____assets_image_png_color__23f4f4f4__}
          title="Image title 2"
        />
        <_components.img
          alt="Image alt text 2"
          meta={__2____assets_image_png_color__23f4f4f4__}
        />
      </div>
      {"\\n"}
      <_components.p>
        {
          "Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
        }
      </_components.p>
      {"\\n"}
      <_components.p>
        <_components.a href="https://example.com" title="Link title">
          <_components.img
            alt="Image alt text 3"
            meta={__3____assets_image_png_reshoot_color__23f4f4f4__}
            title="Image title 3"
          />
        </_components.a>
      </_components.p>
      {"\\n"}
      <_components.p>
        {
          "Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
        }
      </_components.p>
    </>
  );
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = props.components || {};
  return MDXLayout ? (
    <MDXLayout {...props}>
      <_createMdxContent {...props} />
    </MDXLayout>
  ) : (
    _createMdxContent(props)
  );
}
export default MDXContent;

`;

exports[`MDX with @reshoot/remark-mdx-image should correctly transform when removeTrailingLineBreak is disabled 1`] = `
/*@jsxRuntime automatic @jsxImportSource react*/
import __0___assets_image_png__ from "./assets/image.png";
import __1____assets_image_png__ from "../assets/image.png";
import __2____assets_image_png_color__23f4f4f4__ from "../assets/image.png?color=%23f4f4f4";
import __3____assets_image_png_reshoot_color__23f4f4f4__ from "../assets/image.png?reshoot&color=%23f4f4f4";
function _createMdxContent(props) {
  const _components = Object.assign(
    {
      h2: "h2",
      p: "p",
      img: "img",
      a: "a",
    },
    props.components
  );
  return (
    <>
      <_components.h2>{"Hellow World"}</_components.h2>
      {"\\n"}
      <_components.p>
        {
          "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
        }
      </_components.p>
      {"\\n"}
      <_components.img
        src={"https://example.com/image.png?color=%23f4f4f4"}
        alt=""
      />
      {"\\n"}
      {"\\n"}
      {"\\n"}
      <_components.img
        alt="Image alt text 1"
        meta={__0___assets_image_png__}
        title="Image title 1"
      />
      {"\\n"}
      <_components.p>
        {
          "Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat."
        }
      </_components.p>
      {"\\n"}
      <div id="abc">
        <_components.img alt="" meta={__1____assets_image_png__} />
        {"\\n"}
        <_components.img
          alt=""
          meta={__2____assets_image_png_color__23f4f4f4__}
          title="Image title 2"
        />
        {"\\n"}
        <_components.img
          alt="Image alt text 2"
          meta={__2____assets_image_png_color__23f4f4f4__}
        />
      </div>
      {"\\n"}
      <_components.p>
        {
          "Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
        }
      </_components.p>
      {"\\n"}
      <_components.p>
        <_components.a href="https://example.com" title="Link title">
          <_components.img
            alt="Image alt text 3"
            meta={__3____assets_image_png_reshoot_color__23f4f4f4__}
            title="Image title 3"
          />
        </_components.a>
      </_components.p>
      {"\\n"}
      <_components.p>
        {
          "Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
        }
      </_components.p>
    </>
  );
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = props.components || {};
  return MDXLayout ? (
    <MDXLayout {...props}>
      <_createMdxContent {...props} />
    </MDXLayout>
  ) : (
    _createMdxContent(props)
  );
}
export default MDXContent;

`;
